src/main/java/frc/robot/commands/CrocodileDefaultCommand.java

```diff
@@ -3,33 +3,49 @@
 import java.util.function.BooleanSupplier;
 import java.util.function.DoubleSupplier;
 
+import edu.wpi.first.math.filter.Debouncer;
+import edu.wpi.first.wpilibj.XboxController;
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.robot.subsystems.CrocodileSubsystem;
 
 public class CrocodileDefaultCommand extends CommandBase {
     private final CrocodileSubsystem subsystem;
-    private final DoubleSupplier motorSpeed;
-    private final BooleanSupplier slowMovement;
+    private final DoubleSupplier trigger;
+    private final XboxController driverController;
+    private boolean beamBreakTracker = false;
+    private final Debouncer debouncer;
 
-    public CrocodileDefaultCommand(CrocodileSubsystem subsystem, DoubleSupplier motorSpeed,
-            BooleanSupplier slowMovement) {
+    public CrocodileDefaultCommand(CrocodileSubsystem subsystem, DoubleSupplier trigger) {
         this.subsystem = subsystem;
-        this.motorSpeed = motorSpeed;
-        this.slowMovement = slowMovement;
+        this.trigger = trigger;
+        //TODO: get driver controller port
+        //controller to use for rumble
+        driverController = new XboxController(0);
+        debouncer = new Debouncer(0.5);
         addRequirements(subsystem);
     }
 
     // Called every time the scheduler runs while the command is scheduled.
     @Override
     public void execute() {
-        if (slowMovement.getAsBoolean()) {
-            subsystem.setMotorSpeed(.4);
-        } 
-        else if (Math.abs(motorSpeed.getAsDouble()) > 0.05){
-            subsystem.setMotorSpeed(motorSpeed.getAsDouble());
+        long startTime = 0;
+        subsystem.setIntakeSpeed(trigger.getAsDouble());
+        //sets rumble if beam break is broken for 0.5 seconds and is not already rumbling
+        if(debouncer.calculate(subsystem.getBeamBreak()) && !beamBreakTracker){
+            //get current time and set rumble
+            startTime = System.currentTimeMillis();
+            driverController.setRumble(XboxController.RumbleType.kLeftRumble, 0.5);
+            driverController.setRumble(XboxController.RumbleType.kRightRumble, 0.5);
+            beamBreakTracker = true;
         }
-        else {
-            subsystem.setMotorSpeed(0);
+        if(System.currentTimeMillis() - startTime > 500){
```

Comment:
I don't know if your ifs work the way you want them to here. you only get the current time in milliseconds if there is something in the beam break, AND if that if hasn't already executed. So what will happen is you will set `startTime`, set the rumble and then do the rest of your checks. But then what happens on the next loop? `startTime` will be 0, because it is reset on every loop, and now `beamBreakTracker` is true, and so you will rumble for forever.

Why do you need `beamBreakTracker`? I'm not very familiar with how rumble works, is there a problem with setting rumble when it's already rumbling?

I want you to take a moment and imagine what is happening every loop here for your different cases. Draw it out on paper, that always helps me. This is a difficult issue, and the sequencing in this kind of looping system is hard. If you want, we can sit down and work it out together at the meeting today.

Reply:
I want beam break tracker because I want a short rumble each time a new game piece is gotten. I do not want rumble the entire time the game piece is in thr intake. I will look over your suggestions though.